Ds3/Helpers/Jobs/Job.cs

```diff
@@ -187,15 +187,28 @@ private void TransferBlob(IDs3Client client, Blob blob)
                     ChecksumType = _checksumType
                 });
             }
+            catch(Exception ex)
+            {
+                //only closing the stream in the event of exception
+                ContextRangeCloseHelper(blobLength, blob, contextRange => this._streamFactory.CloseStream(contextRange.Context));
+
+                throw ex;
+            }
             finally
             {
                 this._streamFactory.CloseBlob(blob);
             }
 
+            //closing the stream and trgger ItemCompleted event
+            ContextRangeCloseHelper(blobLength, blob, contextRange => this._itemTracker.CompleteRange(contextRange));
+        }
+
+        private void ContextRangeCloseHelper(long blobLength, Blob blob, Action<ContextRange<TItem>> block)
+        {
```

Comment:
Maybe call block something else like contextHandlerFunction?  Block looked to me at first glance like a thread synchronization call.